docs/msbuild/xmlpoke-task.md

```diff
@@ -29,7 +29,7 @@ Sets values as specified by an XPath query into an XML file.
   
 |Parameter|Description|
 |---------------|-----------------|
-|`Namespaces`|Optional `String` parameter.<br /><br /> Specifies the namespaces for XPath query prefixes.|
+|`Namespaces`|Optional `String` parameter.<br /><br /> Specifies the namespaces for XPath query prefixes. Value specifies list of `Namespace` elements with attributes `Prefix` and `Uri`. The attribute `Prefix` specifies the prefix to associate with the namespace specified in `Uri` attribute. Avoid using empty `Prefix` for default namespace, associate it with some artificial prefix and modify query according to it. This parameter must be XML escaped, so replace &lt; with &amp;lt; and &gt; with &amp;gt;.|
```

Comment:
I think I'd say something like

> `Namespaces` is an XML snippet consisting of `Namespace` elements with attributes `Prefix` and `Uri`. The attribute `Prefix` specifies the prefix to associate with the namespace specified in `Uri` attribute. Do not use an empty `Prefix`.

The XML escaping I'm less sure of. It needs to be escaped . . . sometimes. You can store the XML in a property to avoid the escaping, for example

```xml
<Project>
<PropertyGroup>
    <Namespace>
        <Namespace Prefix="dn" Uri="http://schemas.microsoft.com/appx/manifest/foundation/windows10" />
        <Namespace Prefix="mp" Uri="http://schemas.microsoft.com/appx/2014/phone/manifest" />
        <Namespace Prefix="uap" Uri="http://schemas.microsoft.com/appx/manifest/uap/windows10" />
    </Namespace>
</PropertyGroup>

<Target Name="Poke">
  <XmlPoke
    XmlInputPath="Sample.xml"
    Value="MyId"
    Query="/dn:Package/mp:PhoneIdentity/@PhoneProductId"
    Namespaces="$(Namespace)"/>
</Target>
</Project>
```

is equivalent to your example, but more readable.

Reply:
I would go with Rainer's suggestion. If more needs to be said, I'd tend to add it to the Remarks instead: "For the Namespaces parameter, you may need to either store the XML in a property group or include XML escaping. Here's an example with the XML stored in a property group:
<previous code example>

Reply:
And then you could add Rainers example.